internal/controllers/machine/machine_controller.go

```diff
@@ -670,8 +670,19 @@ func (r *Reconciler) drainNode(ctx context.Context, cluster *clusterv1.Cluster,
 	}
 
 	if noderefutil.IsNodeUnreachable(node) {
-		// When the node is unreachable and some pods are not evicted for as long as this timeout, we ignore them.
-		drainer.SkipWaitForDeleteTimeoutSeconds = 60 * 5 // 5 minutes
+		// Kubelet is unreachable, pods will never disappear.
+
+		// SkipWaitForDeleteTimeoutSeconds ensures the drain completes
+		// even if pod objects are not deleted.
+		drainer.SkipWaitForDeleteTimeoutSeconds = 1
+
+		// We don't need to wait for the grace period specified by pods
+		// as we know this won't happen.
+		// The grace period generally won't change anything here as
+		// the kubelet should not be reachable, but it doesn't hurt setting it.
+		drainer.GracePeriodSeconds = 1
```

Comment:
```go
		// The grace period generally won't change anything here as
		// the kubelet should not be reachable, but it doesn't hurt setting it.
```

I'm a bit confused. If it doesnt change anything, why are we setting it?

Reply:
The main reason I added this is that it aligns with [OpenShift MAPI](https://github.com/openshift/machine-api-operator/blob/dcf1387cb69f8257345b2062cff79a6aefb1f5d9/pkg/controller/machine/drain_controller.go#L164-L171). I was not able to think of an edge case this covers, but it also has no negative effect.

As mentioned though, we could also change the value to `0` for it to actually have the effect of pods being force deleted by the kube-apiserver instead of deletion being skipped. Although I was not able to find the behavior of `0` documented, so that might need some additional thorough digging in kube-apiserver. 

@enxebre any thoughts on this? I'm happy to reach out to the MAPI folks to see if we're missing some reason for the value to be set to `1` there.

Reply:
It would be good to have a reason to set this. Otherwise we have no way to reason about changing or not changing this value going forward.

Reply:
Btw nothing against OpenShift MAPI, but I don't want to make changes to CAPI just to align to OpenShift.

Reply:
The kas sets the deletion timestamp to a future date compute the grace period [1].
So effectively we wait SkipWaitForDeleteTimeoutSeconds + GracePeriodSeconds, hence setting it to 1 makes sense to keep it short. (0 would also work but I'd prefer 1 for alignment).

[1] https://github.com/kubernetes/kubernetes/blob/5bf1e95541d90e37f6c6637b5b45d8783e7907aa/staging/src/k8s.io/apiserver/pkg/registry/rest/delete.go#L150

This is where we added to mapi https://github.com/openshift/machine-api-operator/pull/780
This is the driving bug https://bugzilla.redhat.com/show_bug.cgi?id=1905709
These are related features
https://github.com/kubernetes/kubernetes/pull/94381
https://github.com/kubernetes/kubernetes/pull/91342

So maybe just update the comment to reflect the above. cc @sbueringer